src/test/ReverseTest.java

```diff
@@ -0,0 +1,27 @@
+import com.company.reverse.Reverse;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+class ReverseTest {
+   private Reverse reverse = new Reverse();
+
+    @Test
+    void shouldReverseWordWithDigitsUnchanged() {
+        assertEquals("dcba1", reverse.getReversed("abcd1"));
+    }
+
+    @Test
+    void shouldReplaceDigitsSpacesSymbols() {
+        assertEquals("!b1 !A2", reverse.getReversed("!A1 !b2"));
+    }
+
+    @Test
+    void shouldThrowIllegalArgumentException() {
```

Comment:
please add condition under which this exception can be thrown.
In your case we can name it `shouldThrowIllegalArgumentExceptionIfNull`

Reply:
Done